gh_actions_exporter/cost.py

```diff
@@ -0,0 +1,123 @@
+        return 'github-hosted'
+    
+    def relabel_job_labels(self, relabel: Relabel, labels: List[str]) -> Dict[str, str or None]:
+        result: dict = {
+            relabel.label: relabel.default
+        }
+        for label in relabel.values:
+            if label in labels:
+                result[relabel.label] = label
+        return result
+
+    def job_request_labels(self, job_request: dict, webhook: WebHook, settings: Settings) -> dict:
+        labels: dict = dict(
+            runner_type=self.runner_type_job(job_request),
+            job_name=job_request["name"],
+            repository_visibility=webhook.workflow_run.repository.private,
+            repository=webhook.workflow_run.repository.full_name,
+        )
+        for relabel in settings.job_relabelling:
+            if relabel.type == RelabelType.label:
+                labels.update(self.relabel_job_labels(relabel, job_request["labels"]))
+            elif relabel.type == RelabelType.name:
+                labels.update(self.relabel_job_names_dict(relabel, job_request))
+        return labels
+
+    def _get_job_cost(self, job_request: dict, webhook: WebHook) -> float:
+        labels: dict = self.job_request_labels(job_request, webhook, self.settings)
+        flavor = labels.get(self.settings.flavor_label, None)
+        cost_per_min: float = self.settings.job_costs.get(flavor, self.settings.default_cost)
+        duration: float = (job_request["completed_at"].timestamp()
+                    - job_request["started_at"].timestamp())
+
+        return duration / 60 * cost_per_min
+
+    def get_previous_check_run(self, repository: Repository, webhook: WebHook) -> str:
+        url: str = f"/repos/{webhook.repository.full_name}/commits/{webhook.workflow_run.head_sha}/check-runs"
+        check_runs: dict = repository._requester.requestJsonAndCheck("GET", url, {"Accept": "application/vnd.github.v3+json",})[1]
+
+        for check_run in check_runs["check_runs"]:
+            if check_run["name"] == "self.settings.title" and check_run["app"]["id"] == self.settings.github_app_id:
+                return check_run["output"]["summary"]
+        return ""
+
+    def upload_check_run(self, summary: str, webhook: WebHook, access_token: str) -> None:
+        g: Github = Github(login_or_token=access_token)
+        repo: Repository = g.get_repo(webhook.repository.full_name)
+
+        repo.create_check_run(
+            name="Cost",
+            head_sha=webhook.workflow_run.head_sha,
+            status="completed",
+            conclusion="success",
+            output={
+                "title": self.settings.title,
+                "summary": summary
+            }
+        )
+
+    def display_cost(self, webhook: WebHook):
+        if webhook.workflow_run.conclusion:
+            token_generator: TokenGenerator = TokenGenerator(self.settings)
+            access_token: str = token_generator.generate_token()
+            
+            g: Github = Github(access_token)
```

Comment:
Authentication to GitHub should be done with the appropriate method provided by the class. 

Using the `AppAuthentication` class. Example:
```python
  Github(
    app_auth=AppAuthentication(
        app_id=self.settings.github_app_id,
        private_key=self.settings.github_private_pem,
        installation_id=github_app_installation_id,
    ),
)
```
https://github.com/PyGithub/PyGithub/blob/300c5015d41236eb410b9cbe895258294e3500af/tests/Authentication.py#L55

Reply:
Done